phylofisher/utilities/aa_recoder.py

```diff
@@ -0,0 +1,190 @@
+        D6 = {'0': ['A', 'G', 'P', 'S', 'T'],
+            '1': ['D', 'E', 'N', 'Q'],
+            '2': ['H', 'K', 'R'],
+            '3': ['M', 'I', 'L', 'V'],
+            '4': ['W', 'F', 'Y'],
+            '5': ['C']
+            },
+
+
+        # Dayhoff 9-state coding
+        D9 = {'0': ['D', 'E', 'H', 'N', 'Q'],
+                '1': ['I', 'L','M','V'],
+                '2': ['F', 'Y'],
+                '3': ['A', 'S', 'T'],
+                '4': ['K', 'R'],
+                '5': ['G'],
+                '6': ['P'],
+                '7': ['C'],
+                '8': ['W']
+                },
+
+        # Dayhoff 12-state coding
+        D12 = {'0': ['D', 'E', 'Q'],
+                '1': ['M', 'L', 'I', 'V'],
+                '2': ['F', 'Y'],
+                '3': ['K', 'H', 'R'],
+                '4': ['G'],
+                '5': ['A'],
+                '6': ['P'],
+                '7': ['S'],
+                '8': ['T'],
+                '9': ['N'],
+                'A': ['W'],
+                'B': ['C']
+        },
+
+        # Dayhoff 15-state coding
+        D15 = {'0': ['D', 'E', 'Q'],
+                '1': ['M', 'L'],
+                '2': ['I', 'V'],
+                '3': ['F', 'Y'],
+                '4': ['G'],
+                '5': ['A'],
+                '6': ['P'],
+                '7': ['S'],
+                '8': ['T'],
+                '9': ['N'],
+                'A': ['K'],
+                'B': ['H'],
+                'C': ['R'],
+                'D': ['W'],
+                'E': ['C']
+        },
+
+        # Dayhoff 18-state coding
+        D18 = {'0': ['M', 'L'],
+                '1': ['F', 'Y'],
+                '2': ['I'],
+                '3': ['V'],
+                '4': ['G'],
+                '5': ['A'],
+                '6': ['P'],
+                '7': ['S'],
+                '8': ['T'],
+                '9': ['D'],
+                'A': ['E'],
+                'B': ['Q'],
+                'C': ['N'],
+                'D': ['K'],
+                'E': ['H'],
+                'F': ['R'],
+                'G': ['W'],
+                'H': ['C']
+        }
+    )
+
+    out_dict = {'fasta'         : 'fas',
+                'phylip'        : 'phy',
+                'phylip-relaxed': 'phy',
+                'nexus'         : 'nex'}
+
+    dict_strat = args.recoding_strat
+    key = key[dict_strat]
+
+    strats = ['SR4', 'SR6', 'KGB6', 'D6', 'D9', 'D12', 'D15', 'D18']
+
+    
+
+    # Opens input and output files
+    with open(args.input, 'r') as infile, open(f'{args.output}.{out_dict[args.out_format]}', 'w') as outfile:
+        all_records = []
+
+        #check input recoding strategy is valid
+        for in_strat in args.recoding_strat:
+            if in_strat in strats:
+                continue
+            else:
+                break
+            print("Error: Invalid input recoding stragety \n Valid strateies include SR4, SR6, KGB6, Dayhoff6, Dayhoff9, Dayhoff12, Dayhoff15, and Dayhoff18")
```

Comment:
Should the Dayhoff strategies be written as D6, D9 . . . etc in this error message since those are the actual options a user would provide the script?